org.adempiere.base/src/org/compiere/wf/MWFNextCondition.java

```diff
@@ -342,5 +364,40 @@ public MWFNextCondition markImmutable() {
 		makeImmutable();
 		return this;
 	}
+	
+	/**
+	 * Executes SQLStatement of Record and returns result as Object
+	 * @param Record_ID
+	 * @return SQL statement execution result
+	 * @throws Exception
+	 */
+	private Object getSqlResult(int Record_ID) throws Exception {
+		String sqlStatement = (String) get_Value("SQLStatement");
+		String tableName = getAD_Column().getAD_Table().getTableName();

```

Comment:
This construction is very bad for performance
It is always better to use the cached objects from table and column to obtain the information.

Reply:
You mean that I need to use M-class for that?